src/components/modes/teacher/BackgroundToggle.js

```diff
@@ -43,24 +43,26 @@ const BackgroundToggle = () => {
     MUTATION_KEYS.PATCH_APP_SETTING,
   );
 
-  const [ backgroundToggleSetting, setBackgroundToggleSetting ] = useState(null);
-  
+  const [backgroundToggleSetting, setBackgroundToggleSetting] = useState(null);
+
   const { data: appSettings, isSuccess, isLoading } = useAppSettings();
 
   useEffect(() => {
-    if(isSuccess) {
+    if (isSuccess) {
       const backgroundSetting = appSettings?.find(
         ({ name }) => name === APP_SETTINGS.BACKGROUND,
       );
-      if(backgroundSetting){
-        setBackgroundToggleSetting(appSettings?.find(
-          ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
-        ) || DEFAULT_BACKGROUND_TOGGLE);
+      if (backgroundSetting) {
+        setBackgroundToggleSetting(
+          appSettings?.find(
+            ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
+          ) || DEFAULT_BACKGROUND_TOGGLE,
+        );
       }
     }
   }, [appSettings, isSuccess]);
 
-  const toggleDisabled = isLoading || (backgroundToggleSetting === null);
+  const toggleDisabled = isLoading || backgroundToggleSetting === null;
```

Comment:
`Boolean(isLoading || !backgroundToggleSetting)`
You're case doesn't handle toggleSetting = undefined